worlds/AutoWorld.py

```diff
@@ -497,22 +495,21 @@ def get_pre_fill_items(self) -> List["Item"]:
 
     # these two methods can be extended for pseudo-items on state
     def collect(self, state: "CollectionState", item: "Item") -> bool:
+        if not item.advancement:
+            raise Exception("Collect should no longer be called with non advancements.")
+
         """Called when an item is collected in to state. Useful for things such as progressive items or currency."""
         name = self.collect_item(state, item)
-        if name:
-            state.prog_items[self.player][name] += 1
-            return True
-        return False
+        state.prog_items[self.player][name] += 1
+        return True  # Backwards compatibility, this used to denote whether a progression item was picked up
 
     def remove(self, state: "CollectionState", item: "Item") -> bool:
         """Called when an item is removed from to state. Useful for things such as progressive items or currency."""
         name = self.collect_item(state, item, True)
-        if name:
-            state.prog_items[self.player][name] -= 1
-            if state.prog_items[self.player][name] < 1:
-                del (state.prog_items[self.player][name])
-            return True
-        return False
+        state.prog_items[self.player][name] -= 1
+        if state.prog_items[self.player][name] < 1:
+            del (state.prog_items[self.player][name])
+        return True  # Backwards compatibility, this used to denote whether a progression item was picked up
```

Comment:
this comment should probably be changed because remove doesn't 'pick up' items lol